src/features/search/enums.ts

```diff
@@ -27,6 +28,16 @@ type CategoryCriteria = {
   [category in SearchGroupNameEnumv2]: {
     icon: React.FC<BicolorIconInterface>
     facetFilter: SearchGroupNameEnumv2
+    baseColor?: string
```

Comment:
Ce serait pas mieux un
```suggestion
    baseColor?: ColorsEnum | string
```

pour pouvoir le prendre en charge également ? Au niveau du rendu c'est pareil, mais ça peut faciliter l'utilisation.

Reply:
> Ce serait pas mieux un
> 
> pour pouvoir le prendre en charge également ? Au niveau du rendu c'est pareil, mais ça peut faciliter l'utilisation.

Pas trop d'accord : idéalement, les enums de couleurs ne devrait pas être utilisées en dehors du theme, selon moi

Reply:
J'ai mis optionnel et pareil pour gradients parce qu'on a pas de couleur pour la catégorie NONE - est-ce que j'en mets une par défaut ?

Reply:
@voisinhugo Ca tombe bien, on est dans le thème nan ?
On définit le thème des catégories alors...

Reply:
> @voisinhugo Ca tombe bien, on est dans le thème nan ? On définit le thème des catégories alors...
> 
> Et puis lorsque tu utilises `theme.colors.secondary` le résultat c'est un `ColorsEnum`.

C'est pas un `string` justement qu'on récupère plutôt ?

Reply:
> 



> J'ai mis optionnel et pareil pour gradients parce qu'on a pas de couleur pour la catégorie NONE - est-ce que j'en mets une par défaut ?

@tanguyprache arf c'est dommage...
tu peux pas faire un truc comme ça ? Ca marcherait peut-être ?
```ts
    baseColor: category extends SearchGroupNameEnumv2.NONE ? never : ColorsEnum | string
    gradients: category extends SearchGroupNameEnumv2.NONE ? never : Array<Gradient>
```

Reply:
@tanguyprache Apparemment non :P

<img width="226" alt="CleanShot 2023-06-02 at 15 47 59@2x" src="https://github.com/pass-culture/pass-culture-app-native/assets/114916654/ce499f25-d858-49ce-b7bc-1926f9723bfd">

Reply:
> > J'ai mis optionnel et pareil pour gradients parce qu'on a pas de couleur pour la catégorie NONE - est-ce que j'en mets une par défaut ?
> 
> @tanguyprache arf c'est dommage... tu peux pas faire un truc comme ça ? Ca marcherait peut-être ?
> 
> ```ts
>     baseColor: category extends SearchGroupNameEnumv2.NONE ? never : ColorsEnum | string
>     gradients: category extends SearchGroupNameEnumv2.NONE ? never : Array<Gradient>
> ```

Non ca ne marche pas mais peut être si j'essaye de faire un type qui extends CategoryCriteria avec baseColor et gradients required. Mais après c'est assez lourd d'avoir un type juste pour une catégorie qui n'a pas de couleur

Reply:
@dbenfouzari-pass sinon on peut faire ca : 
`type CategoryCriteria = {
  [category in SearchGroupNameEnumv2]: {
    icon: React.FC<BicolorIconInterface>
    facetFilter: SearchGroupNameEnumv2
    baseColor: category extends SearchGroupNameEnumv2.NONE ? undefined : string
    gradients: category extends SearchGroupNameEnumv2.NONE ? undefined : Array<Gradient>
  }
}

export const CATEGORY_CRITERIA: Partial<CategoryCriteria> = {
  [SearchGroupNameEnumv2.NONE]: {
    icon: categoriesIcons.All,
    facetFilter: SearchGroupNameEnumv2.NONE,
    baseColor: undefined,
    gradients: undefined,
  },`